packages/internal/src/generate/typeDefinitions.ts

```diff
@@ -262,29 +266,88 @@ const runCodegenGraphQL = async (
       namingConvention: 'keep', // to allow camelCased query names
       scalars: {
         // We need these, otherwise these scalars are mapped to any
-        // @TODO is there a way we can use scalars defined in packages/graphql-server/src/rootSchema.ts
+        // @TODO is there a way we can use scalars defined in
+        // packages/graphql-server/src/rootSchema.ts
         BigInt: 'number',
         DateTime: 'string',
         Date: 'string',
         JSON: 'Record<string, unknown>',
         JSONObject: 'Record<string, unknown>',
         Time: 'string',
       },
-      omitOperationSuffix: true, // prevent type names being PetQueryQuery, RW generators already append Query/Mutation/etc
+      // prevent type names being PetQueryQuery, RW generators already append
+      // Query/Mutation/etc
+      omitOperationSuffix: true,
     },
     generates,
     silent: false,
     errorsOnly: true,
   }
 
+  return codegenConfig
+}
+
+export const generateTypeDefGraphQLWeb = async () => {
+  const rwjsPaths = getPaths()
+  const documentsGlob = './web/src/**/!(*.d).{ts,tsx,js,jsx}'
+
+  const generates = {
+    [path.join(rwjsPaths.web.types, 'graphql.d.ts')]: {
+      documents: documentsGlob,
+      plugins: [
+        {
+          typescript: {
+            enumsAsTypes: true,
+          },
+        },
+        'typescript-operations',
+      ],
+    },
+  }
+
+  const options = generateLoadTypedefsConfig(generates)
+
+  try {
+    await loadDocuments([documentsGlob], options)
+  } catch {
+    // No GraphQL documents present (web), no need to try to run codegen
+    return []
+  }
+
+  try {
+    return await runCodegenGraphQL(generates)
+  } catch {
+    console.error()
+    console.error('Error: Could not generate GraphQL type definitions (web)')
+    console.error()
+    return []
+  }
+}
+
+/**
+ * This is the function used internally by generateTypeDefGraphQLApi and generateTypeDefGraphQLWeb
+ * And contains the base configuration for generating gql types with codegen
+ *
+ * Named a little differently to make it easier to spot
+ */
+
+// CLI generate also takes cwd
+type CodegenConfig = CodegenTypes.Config & { cwd: string }
+
+const runCodegenGraphQL = async (
+  generates: Record<string, CodegenTypes.ConfiguredOutput>
+) => {
+  const rwjsPaths = getPaths()
+  type GenerateResponse = { filename: string; contents: string }[]
+
   const userCodegenConfig = await loadCodegenConfig({
     configFilePath: rwjsPaths.base,
   })
 
   // https://www.graphql-code-generator.com/docs/getting-started/programmatic-usage#using-the-cli-instead-of-core
   const f: GenerateResponse = await generate(
```

Comment:
Since you're loading in the documents, I think we should use `generate` from `@graphql-codegen/core` rather than the CLI - otherwise we're just loading all the CLI stuff for no reason.

Reply:
I didn't want to change this. Wanted to limit the scope of this PR. As I mentioned in the PR description I did test to make sure I didn't make this function too much slower by my changes.

But if you still want me to make this change in this PR, just tell me (again) and I'll do it.

Reply:
Hmmm.. I think when you use the cli, it loads a bunch of libs underneath, that help with the parsing! I think we might get a performance _advantage_ by using codegen/core. The reason I (or maybe Peter) originally used the CLI was just for the parsing bit - at the time I didn't understand the power of `@graphql/tools` :) 

I would love for you to try, and I will definitely help validate that everything works as expected, but if it seems like too much work, we can always decide to keep it as is with the cli.

Reply:
Cool. I'll spend a few hours on it.

If it ends up taking too long, can we get this PR merged and then table the cli->core refactor for later?

Are the new names I switched to acceptable?

Reply:
> Are the new names I switched to acceptable?

Super clear! Thank youuu


> Cool. I'll spend a few hours on it.

If you're short on time, I can take over and give it a go, just say the word

Reply:
> If you're short on time, I can take over and give it a go, just say the word

You'll have to pry this one out of my cold dead hands! 😁 

OK, that might have been a bit excessive 😛 
But I enjoy working with this, so I'd love to give it a go 🙂

Reply:
https://www.graphql-code-generator.com/docs/advanced/programmatic-usage - your essential guide!